### ISSUE TITLE
False positive S113 for httpx.AsyncClient

### ISSUE THREAD

**Issue by dybi** [astral-sh/ruff#12240]
`S113` is reported by `ruff` for `httpx.AsyncClient()` (https://docs.astral.sh/ruff/rules/request-without-timeout/)
whereas `httpx` works differntly than `requests` in this area and there is a finite default: https://github.com/encode/httpx/blob/db9072f998b53ff66d50778bf5edee8e2cc8ede1/httpx/_client.py#L1390

Minimal app do reproduce:
```python
import httpx

client = httpx.AsyncClient()
```

`ruff` version = `0.5.1.`

**Comment by MichaReiser** [comment]
Thanks for reporting! I think this is already fixed and will go out in our next release https://github.com/astral-sh/ruff/pull/12213
